<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exon-Intron Split Analysis (EISA) in R • eisaR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Exon-Intron Split Analysis (EISA) in R">
<meta property="og:description" content="Exon-intron split analysis (EISA) uses ordinary RNA-seq data to measure
    changes in mature RNA and pre-mRNA reads across different experimental conditions
    to quantify transcriptional and post-transcriptional regulation of gene expression.
    For details see Gaidatzis et al., Nat Biotechnol 2015. doi: 10.1038/nbt.3269.
    eisaR implements the major steps of EISA in R.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">eisaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/eisaR.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/eisaR.html">eisaR</a>
    </li>
    <li>
      <a href="articles/rna-velocity.html">Feature extraction for RNA velocity</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fmicompbio/eisaR">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<p><img src="inst/www/eisaR.png" align="right" alt="" width="120"></p>
<div id="eisar-exon-intron-split-analysis-eisa-in-r" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#eisar-exon-intron-split-analysis-eisa-in-r" class="anchor"></a><code>eisaR</code>: Exon-Intron Split Analysis (EISA) in R</h1></div>
<p><br></p>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Exon-intron split analysis (<code>EISA</code>) uses ordinary RNA-seq data to measure changes in mature RNA and pre-mRNA reads across different experimental conditions to quantify transcriptional and post-transcriptional regulation of gene expression.</p>
<p>For details see Gaidatzis et al., Nat Biotechnol 2015. doi: 10.1038/nbt.3269.<br><code>eisaR</code> implements the major steps of EISA in R. In addition, it contains functionality for extracting spliced and unspliced transcript sequences, as well as intron sequences (with similar options as the <a href="https://github.com/BUStools/BUSpaRse" class="external-link">BUSpaRse</a>) package), from an annotated genome. These sequences can be indexed and used, e.g., for quantification in preparation for RNA velocity estimation.</p>
<p>Developed by:</p>
<ul>
<li><a href="https://github.com/mbstadler" class="external-link">Michael Stadler</a></li>
<li>Dimos Gaidatzis</li>
<li><a href="https://github.com/LukasBurger" class="external-link">Lukas Burger</a></li>
<li><a href="https://github.com/csoneson" class="external-link">Charlotte Soneson</a></li>
</ul>
<p>Also a big “thank you” for contributions to:</p>
<ul>
<li><a href="https://github.com/federicomarini" class="external-link">Federico Marini</a></li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install <code>eisaR</code> from Bioconductor, you will need at least R 4.0 and Bioconductor 3.11, which is available on April 28, 2020. Then use the following within R (see also <a href="https://bioconductor.org/packages/eisaR/" class="external-link">eisaR download page</a>):</p>
<pre><code># BiocManager is needed to install Bioconductor packages
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

# Install eisaR
BiocManager::install("eisaR")</code></pre>
</div>
<div id="functionality" class="section level2">
<h2 class="hasAnchor">
<a href="#functionality" class="anchor"></a>Functionality</h2>
<p>All you need is RNA-seq data from at least two conditions (e.g. wildtype and mutant). The <code>eisaR</code> package contains convenience functions to facilitate the steps in an exon-intron split analysis, which consists of:</p>
<ol style="list-style-type: decimal">
<li>preparing the annotation (exonic and gene body coordinate ranges)<br>
</li>
<li>quantifying RNA-seq alignments in exons and introns<br>
</li>
<li>calculating and comparing exonic and intronic changes across conditions<br>
</li>
<li>visualizing the results</li>
</ol>
<p>For the steps 1. and 2. above, this <code>eisaR</code> vignette makes use of Bioconductor annotation and the <a href="https://bioconductor.org/packages/QuasR/" class="external-link">QuasR</a> package. It is also possible to obtain count tables for exons and introns using some other pipeline or approach, and directly start with step 3.</p>
</div>
<div id="reference" class="section level2">
<h2 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h2>
<p><code>EISA</code> has been described in:</p>
<p>“Analysis of intronic and exonic reads in RNA-seq data characterizes transcriptional and post-transcriptional regulation.”<br>
Gaidatzis D., Burger L., Florescu M. and Stadler, M.B.<br><em>Nat Biotechnol.</em> <strong>2015</strong>; 33(7):722-9. <a href="https://www.ncbi.nlm.nih.gov/pubmed/26098447" class="external-link">PubMed: 26098447</a>, <a href="https://doi.org/10.1038/nbt.3269" class="external-link">doi: 10.1038/nbt.3269</a></p>
<p>The functionality for extracting spliced and unspliced transcript sequences, as well as intron sequences, for use in RNA velocity analysis, has been described in:</p>
<p>“Preprocessing choices affect RNA velocity results for droplet scRNA-seq data.” Soneson C., Srivastava A., Patro R. and Stadler, M.B. <em>bioRxiv</em> <strong>2020</strong>. <a href="https://www.biorxiv.org/content/10.1101/2020.03.13.990069v1" class="external-link">bioRxiv</a>, <a href="https://doi.org/10.1101/2020.03.13.990069" class="external-link">doi: 10.1101/2020.03.13.990069</a></p>
</div>
<div id="software-status" class="section level2">
<h2 class="hasAnchor">
<a href="#software-status" class="anchor"></a>Software status</h2>
<table class="table">
<colgroup>
<col width="25%">
<col width="25%">
<col width="25%">
<col width="25%">
</colgroup>
<thead><tr class="header">
<th align="center">Platforms</th>
<th align="center">OS</th>
<th align="center">R CMD check</th>
<th align="center">Coverage</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">GitHub Actions</td>
<td align="center">Linux/Windows/macOS</td>
<td align="center"><a href="https://github.com/fmicompbio/eisaR/actions" class="external-link"><img src="https://github.com/fmicompbio/eisaR/workflows/R-CMD-check/badge.svg" alt="R build status"></a></td>
<td align="center"><a href="https://codecov.io/github/fmicompbio/eisaR" class="external-link"><img src="https://codecov.io/github/fmicompbio/eisaR/coverage.svg?branch=master" alt="Codecov.io coverage status"></a></td>
</tr>
<tr class="even">
<td align="center">Bioc (<a href="http://bioconductor.org/packages/devel/bioc/html/eisaR.html" class="external-link"><em>devel</em></a>)</td>
<td align="center">Multiple</td>
<td align="center"><a href="http://bioconductor.org/checkResults/devel/bioc-LATEST/eisaR" class="external-link"><img src="http://bioconductor.org/shields/build/devel/bioc/eisaR.svg" alt="Bioconductor-devel Build Status"></a></td>
<td align="center"><a href="https://codecov.io/github/fmicompbio/eisaR" class="external-link"><img src="https://codecov.io/github/fmicompbio/eisaR/coverage.svg?branch=master" alt="Codecov.io coverage status"></a></td>
</tr>
<tr class="odd">
<td align="center">Bioc (<a href="http://bioconductor.org/packages/release/bioc/html/eisaR.html" class="external-link"><em>release</em></a>)</td>
<td align="center">Multiple</td>
<td align="center"><a href="http://bioconductor.org/checkResults/release/bioc-LATEST/eisaR" class="external-link"><img src="http://bioconductor.org/shields/build/release/bioc/eisaR.svg" alt="Bioconductor-release Build Status"></a></td>
<td align="center"><a href="https://codecov.io/github/fmicompbio/eisaR" class="external-link"><img src="https://codecov.io/github/fmicompbio/eisaR/coverage.svg?branch=master" alt="Codecov.io coverage status"></a></td>
</tr>
</tbody>
</table>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/eisaR">https://​www.bioconductor.org/​packages/​eisaR</a>
</li>
<li>Browse source code at <br><a href="https://github.com/fmicompbio/eisaR/">https://​github.com/​fmicompbio/​eisaR/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/fmicompbio/eisaR/issues">https://​github.com/​fmicompbio/​eisaR/​issues</a>
</li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing eisaR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://github.com/mbstadler">Michael Stadler</a> <br><small class="roles"> Author, maintainer </small>  </li>
<li>Dimos Gaidatzis <br><small class="roles"> Author </small>  </li>
<li>
<a href="https://github.com/LukasBurger">Lukas Burger</a> <br><small class="roles"> Author </small>  </li>
<li>
<a href="https://csoneson.github.io">Charlotte Soneson</a> <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/mbstadler" class="external-link">Michael Stadler</a>, Dimos Gaidatzis, <a href="https://github.com/LukasBurger" class="external-link">Lukas Burger</a>, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
