<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exon-Intron Split Analysis (EISA) in R • eisaR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Exon-Intron Split Analysis (EISA) in R">
<meta property="og:description" content="Exon-intron split analysis (EISA) uses ordinary RNA-seq data to measure
    changes in mature RNA and pre-mRNA reads across different experimental conditions
    to quantify transcriptional and post-transcriptional regulation of gene expression.
    For details see Gaidatzis et al., Nat Biotechnol 2015. doi: 10.1038/nbt.3269.
    eisaR implements the major steps of EISA in R.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">eisaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/eisaR.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/eisaR.html">eisaR</a>
    </li>
    <li>
      <a href="articles/rna-velocity.html">Feature extraction for RNA velocity</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fmicompbio/eisaR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="eisar-exon-intron-split-analysis-eisa-in-r" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#eisar-exon-intron-split-analysis-eisa-in-r" class="anchor"></a><code>eisaR</code>: Exon-Intron Split Analysis (EISA) in R</h1></div>
<p><br></p>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Exon-intron split analysis (<code>EISA</code>) uses ordinary RNA-seq data to measure changes in mature RNA and pre-mRNA reads across different experimental conditions to quantify transcriptional and post-transcriptional regulation of gene expression.</p>
<p>For details see Gaidatzis et al., Nat Biotechnol 2015. doi: 10.1038/nbt.3269.<br>
eisaR implements the major steps of EISA in R. In addition, it contains functionality for extracting spliced and unspliced transcript sequences, as well as intron sequences (with similar options as the <a href="https://github.com/BUStools/BUSpaRse">BUSpaRse</a>) package), from an annotated genome. These sequences can be indexed and used, e.g., for quantification in preparation for RNA velocity estimation.</p>
<p>Developed by:</p>
<ul>
<li><a href="https://github.com/mbstadler">Michael Stadler</a></li>
<li>Dimos Gaidatzis</li>
<li><a href="https://github.com/LukasBurger">Lukas Burger</a></li>
<li><a href="https://github.com/csoneson">Charlotte Soneson</a></li>
</ul>
<p>Also a big “thank you” for contributions to:</p>
<ul>
<li><a href="https://github.com/federicomarini">Federico Marini</a></li>
</ul>
</div>
<div id="functionality" class="section level2">
<h2 class="hasAnchor">
<a href="#functionality" class="anchor"></a>Functionality</h2>
<p>All you need is RNA-seq data from at least two conditions (e.g. wildtype and mutant). The <code>eisaR</code> package contains convenience functions to facilitate the steps in an exon-intron split analysis, which consists of:</p>
<ol>
<li>preparing the annotation (exonic and gene body coordinate ranges)<br>
</li>
<li>quantifying RNA-seq alignments in exons and introns<br>
</li>
<li>calculating and comparing exonic and intronic changes across conditions<br>
</li>
<li>visualizing the results</li>
</ol>
<p>For the steps 1. and 2. above, this <code>eisaR</code> vignette makes use of Bioconductor annotation and the <a href="https://bioconductor.org/packages/QuasR/">QuasR</a> package. It is also possible to obtain count tables for exons and introns using some other pipeline or approach, and directly start with step 3.</p>
</div>
<div id="reference" class="section level2">
<h2 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h2>
<p><code>EISA</code> has been described in:</p>
<p>“Analysis of intronic and exonic reads in RNA-seq data characterizes transcriptional and post-transcriptional regulation.”<br>
Gaidatzis D., Burger L., Florescu M. and Stadler, M.B.<br><em>Nat Biotechnol.</em> <strong>2015</strong>; 33(7):722-9. <a href="https://www.ncbi.nlm.nih.gov/pubmed/26098447">PubMed: 26098447</a>, <a href="https://doi.org/10.1038/nbt.3269">doi: 10.1038/nbt.3269</a></p>
</div>
<div id="software-status" class="section level2">
<h2 class="hasAnchor">
<a href="#software-status" class="anchor"></a>Software status</h2>
<table class="table">
<thead><tr class="header">
<th align="center">Platforms</th>
<th align="center">OS</th>
<th align="center">R CMD check</th>
<th align="center">Coverage</th>
</tr></thead>
<tbody><tr class="odd">
<td align="center">Travis CI</td>
<td align="center">Linux</td>
<td align="center"><a href="https://travis-ci.com/fmicompbio/eisaR"><img src="https://travis-ci.com/fmicompbio/eisaR.svg?branch=master" alt="Travis CI build status"></a></td>
<td align="center"><a href="https://codecov.io/github/fmicompbio/eisaR"><img src="https://codecov.io/github/fmicompbio/eisaR/coverage.svg?branch=master" alt="Codecov.io coverage status"></a></td>
</tr></tbody>
</table>
<!--
## Download from Bioconductor
[QuasR download page](https://bioconductor.org/packages/QuasR/)

## Software status

| Platforms |  OS  | R CMD check | Coverage | 
|:----------------:|:----------------:|:----------------:|:----------------:|
| Travis CI | Linux | [![Travis CI build status](https://travis-ci.com/fmicompbio/QuasR.svg?branch=master)](https://travis-ci.com/fmicompbio/QuasR) | [![Codecov.io coverage status](https://codecov.io/github/fmicompbio/QuasR/coverage.svg?branch=master)](https://codecov.io/github/fmicompbio/QuasR) |
| Bioc ([_devel_](http://bioconductor.org/packages/devel/bioc/html/QuasR.html)) | Multiple | [![Bioconductor-devel Build Status](http://bioconductor.org/shields/build/devel/bioc/QuasR.svg)](http://bioconductor.org/checkResults/devel/bioc-LATEST/QuasR) | `NA` |
| Bioc ([_release_](http://bioconductor.org/packages/release/bioc/html/QuasR.html)) | Multiple | [![Bioconductor-release Build Status](http://bioconductor.org/shields/build/release/bioc/QuasR.svg)](http://bioconductor.org/checkResults/release/bioc-LATEST/QuasR) | `NA` |
-->
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing eisaR</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://github.com/mbstadler">Michael Stadler</a> <br><small class="roles"> Author, maintainer </small>  </li>
<li>Dimos Gaidatzis <br><small class="roles"> Author </small>  </li>
<li>
<a href="https://github.com/LukasBurger">Lukas Burger</a> <br><small class="roles"> Author </small>  </li>
<li>
<a href="https://csoneson.github.io">Charlotte Soneson</a> <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/mbstadler">Michael Stadler</a>, Dimos Gaidatzis, <a href="https://github.com/LukasBurger">Lukas Burger</a>, <a href="https://csoneson.github.io">Charlotte Soneson</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
