<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate a GRangesList object with feature ranges — getFeatureRanges • eisaR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate a GRangesList object with feature ranges — getFeatureRanges" />
<meta property="og:description" content="Generate a GRangesList object with genomic ranges for (any combination of)
spliced transcripts, unspliced transcripts and introns." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">eisaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/eisaR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/eisaR.html">eisaR</a>
    </li>
    <li>
      <a href="../articles/rna-velocity.html">Feature extraction for RNA velocity</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/eisaR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a GRangesList object with feature ranges</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/eisaR/blob/master/R/getFeatureRanges.R'><code>R/getFeatureRanges.R</code></a></small>
    <div class="hidden name"><code>getFeatureRanges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generate a GRangesList object with genomic ranges for (any combination of)
spliced transcripts, unspliced transcripts and introns.</p>
    </div>

    <pre class="usage"><span class='fu'>getFeatureRanges</span>(
  <span class='kw'>gtf</span>,
  featureType = <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"spliced"</span>, <span class='st'>"intron"</span>),
  intronType = <span class='st'>"separate"</span>,
  flankLength = <span class='fl'>90L</span>,
  joinOverlappingIntrons = <span class='fl'>FALSE</span>,
  verbose = <span class='fl'>TRUE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gtf</th>
      <td><p>Path to gtf file.</p></td>
    </tr>
    <tr>
      <th>featureType</th>
      <td><p>Character vector indicating the type(s) of features to
extract, any subset of <code><a href='https://rdrr.io/r/base/c.html'>c("spliced", "intron", "unspliced")</a></code>.</p></td>
    </tr>
    <tr>
      <th>intronType</th>
      <td><p>Character vector indicating how to define the introns
(only used if "intron" is part of <code>featureType</code>). Has to be either
"separate" (introns are defined for each transcript separately) or
"collapse" (transcripts of the same gene are first collapsed before
introns are defined as any non-exonic part of the gene locus).</p></td>
    </tr>
    <tr>
      <th>flankLength</th>
      <td><p>Integer scalar indicating the length of the flanking
sequence added to each side of each extracted intron (only used if
"intron" is included in <code>featureType</code>).</p></td>
    </tr>
    <tr>
      <th>joinOverlappingIntrons</th>
      <td><p>Logical scalar indicating whether two introns
that overlap (after adding the flanking sequences) should be joined into
one feature.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Logical scalar, whether to print out progress messages.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns a <code>GRangesList</code> object where each element represents
one extracted feature. The metadata of this object contains two
<code>data.frame</code>s mapping corresponding identifiers between the
different feature types, as well as a list of all features for each type.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Charlotte Soneson</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>  <span class='co'>## Get feature ranges</span>
  <span class='kw'>grl</span> <span class='op'>&lt;-</span> <span class='fu'>getFeatureRanges</span>(
    gtf = <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata/small_example.gtf"</span>, package = <span class='st'>"eisaR"</span>),
    featureType = <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"spliced"</span>, <span class='st'>"intron"</span>),
    intronType = <span class='st'>"separate"</span>,
    flankLength = <span class='fl'>5L</span>,
    joinOverlappingIntrons = <span class='fl'>FALSE</span>,
    verbose = <span class='fl'>TRUE</span>
  )
</div><div class='output co'>#&gt; <span class='message'>Import genomic features from the file as a GRanges object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Prepare the 'metadata' data frame ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>Make the TxDb object ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:1 mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>Extracting spliced transcript features</span></div><div class='output co'>#&gt; <span class='message'>Extracting introns using the separate approach</span></div><div class='input'>  
  <span class='co'>## GRangesList</span>
  <span class='kw'>grl</span>
</div><div class='output co'>#&gt; GRangesList object of length 17:
#&gt; $tx1.3
#&gt; GRanges object with 1 range and 5 metadata columns:
#&gt;       seqnames    ranges strand |     exon_id exon_rank transcript_id
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;   &lt;character&gt;
#&gt;   [1]     chr1     61-90      + |          E6         1         tx1.3
#&gt;           gene_id        type
#&gt;       &lt;character&gt; &lt;character&gt;
#&gt;   [1]          g1        exon
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $tx1.1
#&gt; GRanges object with 2 ranges and 5 metadata columns:
#&gt;       seqnames    ranges strand |     exon_id exon_rank transcript_id
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;   &lt;character&gt;
#&gt;   [1]     chr1     61-90      + |          E1         1         tx1.1
#&gt;   [2]     chr1   141-160      + |          E2         2         tx1.1
#&gt;           gene_id        type
#&gt;       &lt;character&gt; &lt;character&gt;
#&gt;   [1]          g1        exon
#&gt;   [2]          g1        exon
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; $tx1.2
#&gt; GRanges object with 3 ranges and 5 metadata columns:
#&gt;       seqnames    ranges strand |     exon_id exon_rank transcript_id
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;integer&gt;   &lt;character&gt;
#&gt;   [1]     chr1     61-80      + |          E3         1         tx1.2
#&gt;   [2]     chr1   101-120      + |          E4         2         tx1.2
#&gt;   [3]     chr1   141-160      + |          E5         3         tx1.2
#&gt;           gene_id        type
#&gt;       &lt;character&gt; &lt;character&gt;
#&gt;   [1]          g1        exon
#&gt;   [2]          g1        exon
#&gt;   [3]          g1        exon
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths
#&gt; 
#&gt; ...
#&gt; &lt;14 more elements&gt;</div><div class='input'>  
  <span class='co'>## Corresponding transcript/gene IDs</span>
  <span class='kw'>S4Vectors</span>::<span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html'>metadata</a></span>(<span class='kw'>grl</span>)<span class='op'>$</span><span class='kw'>corrtx</span>
</div><div class='output co'>#&gt;   spliced
#&gt; 1   tx1.3
#&gt; 2   tx1.1
#&gt; 3   tx1.2
#&gt; 4 tx2.1-I
#&gt; 5 tx2.2-U
#&gt; 6   tx3.3
#&gt; 7   tx3.1
#&gt; 8   tx3.2</div><div class='input'>  <span class='kw'>S4Vectors</span>::<span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html'>metadata</a></span>(<span class='kw'>grl</span>)<span class='op'>$</span><span class='kw'>corrgene</span>
</div><div class='output co'>#&gt;   spliced intron
#&gt; 1      g1   g1-I
#&gt; 2    g2-I g2-I-I
#&gt; 3    g3-U g3-U-I</div><div class='input'>  
  <span class='co'>## List of features of different types</span>
  <span class='kw'>S4Vectors</span>::<span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html'>metadata</a></span>(<span class='kw'>grl</span>)<span class='op'>$</span><span class='kw'>featurelist</span>
</div><div class='output co'>#&gt; $spliced
#&gt; [1] "tx1.3"   "tx1.1"   "tx1.2"   "tx2.1-I" "tx2.2-U" "tx3.3"   "tx3.1"  
#&gt; [8] "tx3.2"  
#&gt; 
#&gt; $intron
#&gt; [1] "tx1.1-I"    "tx1.2-I"    "tx1.2-I1"   "tx2.1-I-I"  "tx2.2-U-I" 
#&gt; [6] "tx2.2-U-I1" "tx3.1-I"    "tx3.2-I"    "tx3.2-I1"  
#&gt; </div><div class='input'>  
  <span class='co'>## Get feature sequences</span>
  <span class='co'>if</span> (<span class='fu'><a href='https://rdrr.io/r/base/ns-load.html'>requireNamespace</a></span>(<span class='st'>"BSgenome"</span>, quietly = <span class='fl'>TRUE</span>)) {
    <span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='kw'>BSgenome</span>)
    <span class='kw'>genome</span> <span class='op'>&lt;-</span> <span class='kw'>Biostrings</span>::<span class='fu'><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html'>readDNAStringSet</a></span>(
      <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata/small_example_genome.fa"</span>, package = <span class='st'>"eisaR"</span>))
    <span class='kw'>seqs</span> <span class='op'>&lt;-</span> <span class='kw'>GenomicFeatures</span>::<span class='fu'><a href='https://rdrr.io/pkg/GenomicFeatures/man/extractTranscriptSeqs.html'>extractTranscriptSeqs</a></span>(x = <span class='kw'>genome</span>, 
                                                   transcripts = <span class='kw'>grl</span>)
    <span class='kw'>seqs</span>
  }
</div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biostrings</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: XVector</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘Biostrings’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    strsplit</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: rtracklayer</span></div><div class='output co'>#&gt; DNAStringSet object of length 17:
#&gt;      width seq                                              names               
#&gt;  [1]    30 <span style='color: #000000;background-color: #BBBBBB;'>TGTTCAAGCGCCTACGTGGGTTGGGCAACC</span><span>                   tx1.3
#&gt;  [2]    50 </span><span style='color: #000000;background-color: #BBBBBB;'>TGTTCAAGCGCCTACGTGGGTTG</span><span>...</span><span style='color: #000000;background-color: #BBBBBB;'>CCGACCTAGAGGTAATAAGGGT</span><span> tx1.1
#&gt;  [3]    60 </span><span style='color: #000000;background-color: #BBBBBB;'>TGTTCAAGCGCCTACGTGGGGAG</span><span>...</span><span style='color: #000000;background-color: #BBBBBB;'>GTGACCTAGAGGTAATAAGGGT</span><span> tx1.2
#&gt;  [4]    40 </span><span style='color: #000000;background-color: #BBBBBB;'>CGCTTGAACACGTATAATGCAGTCTCGTCATCTCCCCCAG</span><span>         tx2.1-I
#&gt;  [5]    40 </span><span style='color: #000000;background-color: #BBBBBB;'>CCCACGTAGGCGCTTGAACAGACCACGCGGTCTCCCCCAG</span><span>         tx2.2-U
#&gt;  ...   ... ...
#&gt; [13]    25 </span><span style='color: #000000;background-color: #BBBBBB;'>CGCGGCGGCTAGTCTCGTCATCTCC</span><span>                        tx2.2-U-I
#&gt; [14]    25 </span><span style='color: #000000;background-color: #BBBBBB;'>GAACACGTATAATGCGGCAGGACCA</span><span>                        tx2.2-U-I1
#&gt; [15]    60 </span><span style='color: #000000;background-color: #BBBBBB;'>ATCCACAGTGATAACTTACTTGG</span><span>...</span><span style='color: #000000;background-color: #BBBBBB;'>GTGTTGCCATTTTATCGGCTTG</span><span> tx3.1-I
#&gt; [16]    30 </span><span style='color: #000000;background-color: #BBBBBB;'>CCCGTTGTGCATCCACAGTGATAACTTACT</span><span>                   tx3.2-I
#&gt; [17]    30 </span><span style='color: #000000;background-color: #BBBBBB;'>ATAGAGGAGTGTTGCCATTTTATCGGCTTG</span><span>                   tx3.2-I1</div><div class='input'> 
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://github.com/mbstadler'>Michael Stadler</a>, Dimos Gaidatzis, <a href='https://github.com/LukasBurger'>Lukas Burger</a>, <a href='https://csoneson.github.io'>Charlotte Soneson</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


