<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get exonic/gene body regions from a transcript database. — getRegionsFromTxDb • eisaR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get exonic/gene body regions from a transcript database. — getRegionsFromTxDb"><meta property="og:description" content="From a transcript database package (TxDb),
extract exonic and gene body ranges for use with EISA. These regions can
be used to quantify RNA-seq alignments in exons and gene bodies, respectively.
Intronic counts can then be obtained from the difference between gene bodies
and exonic region counts."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">eisaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.7.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/eisaR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/eisaR.html">eisaR</a>
    </li>
    <li>
      <a href="../articles/rna-velocity.html">Feature extraction for RNA velocity</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/fmicompbio/eisaR" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get exonic/gene body regions from a transcript database.</h1>
    <small class="dont-index">Source: <a href="https://github.com/fmicompbio/eisaR/blob/HEAD/R/getRegionsFromTxDb.R" class="external-link"><code>R/getRegionsFromTxDb.R</code></a></small>
    <div class="hidden name"><code>getRegionsFromTxDb.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>From a transcript database package (<code><a href="https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html" class="external-link">TxDb</a></code>),
extract exonic and gene body ranges for use with EISA. These regions can
be used to quantify RNA-seq alignments in exons and gene bodies, respectively.
Intronic counts can then be obtained from the difference between gene bodies
and exonic region counts.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class="sourceCode r"><code><span class="fu">getRegionsFromTxDb</span><span class="op">(</span><span class="va">txdb</span>, exonExt <span class="op">=</span> <span class="fl">10L</span>, strandedData <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>

    <div id="ref-sections">
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>txdb</dt>
<dd><p>a <code>TxDb</code> or an <code>EnsDb</code> object with the transcript annotations.</p></dd>
<dt>exonExt</dt>
<dd><p><code>numeric</code> (default = 10L). Exonic ranges will be extended
on either side by this many nucleotides, in order to avoid "bleed-over" of
exonic alignments into adjacent intronic regions.</p></dd>
<dt>strandedData</dt>
<dd><p><code>logical(1)</code>. If <code>TRUE</code>, the RNA-seq data is
assumed to be strand-specific, and therefore only overlapping genes that
are on the same strand will be filtered out. If <code>FALSE</code>, also genes
overlapping on opposite strands will be filtered out.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a <code>list</code> with elements "exons" and "genebodies", containing
named <code>GenomicRanges</code> objects with ranges for exons and gene bodies,
respectively.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The exonic regions are generated as follows:</p><ol><li><p>extract exons by gene from the <code>txdb</code></p></li>
<li><p>extend each exon by <code>exonExt</code></p></li>
<li><p>combine overlapping exons within each gene</p></li>
<li><p>create gene body ranges from the most extreme exonic coordinates</p></li>
<li><p>filter out genes that have only a single exon (no intron), have exons
on more than a single chromosome or on both strands, or that
overlap other genes</p></li>
</ol></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html" class="external-link">TxDb</a></code> for details on <code>TxDb</code> objects
and how to create them, e.g. from <code>.gtf</code> files.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Michael Stadler</p>
    </div>
    </div>

    <div id="examples">
    <h2>Examples</h2>
    <div class="ref-examples sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"AnnotationDbi"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">    <span class="va">txdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">loadDb</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hg19sub.sqlite"</span>, package <span class="op">=</span> <span class="st">"eisaR"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">regL</span> <span class="op">&lt;-</span> <span class="fu">getRegionsFromTxDb</span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">regL</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicFeatures</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> extracting exon coordinates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> total number of genes/exons: 12/32</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> removing overlapping/single-exon/ambiguous genes (8)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> creating filtered regions for 4 genes (33.3%) with 20 exons (62.5%)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      exons genebodies </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         20          4 </span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://github.com/mbstadler" class="external-link">Michael Stadler</a>, Dimos Gaidatzis, <a href="https://github.com/LukasBurger" class="external-link">Lukas Burger</a>, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

