<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Generating reference files for spliced and unspliced abundance estimation with alignment-free methods • eisaR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Generating reference files for spliced and unspliced abundance estimation with alignment-free methods">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">eisaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/eisaR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/eisaR.html">eisaR</a>
    </li>
    <li>
      <a href="../articles/rna-velocity.html">Feature extraction for RNA velocity</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fmicompbio/eisaR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Generating reference files for spliced and unspliced abundance estimation with alignment-free methods</h1>
                        <h4 class="author">Charlotte Soneson</h4>
            
            <h4 class="date">2020-03-23</h4>
      
      
      <div class="hidden name"><code>rna-velocity.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In this vignette, we show how to prepare reference files for estimating abundances of spliced and unspliced abundances with alignment-free methods (e.g., <a href="https://salmon.readthedocs.io/en/latest/salmon.html">Salmon</a>, <a href="https://salmon.readthedocs.io/en/latest/alevin.html">alevin</a> or <a href="https://pachterlab.github.io/kallisto/">kallisto</a>). Such abundances are used as input, e.g., for estimation of RNA velocity in single-cell data.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(eisaR)</a></code></pre></div>
</div>
<div id="generate-feature-ranges" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-feature-ranges" class="anchor"></a>Generate feature ranges</h1>
<p>The first step is to generate a <code>GRangesList</code> object containing the genomic ranges for the features of interest (spliced transcripts, and either unspliced transcripts or intron sequences). This is done using the <code><a href="../reference/getFeatureRanges.html">getFeatureRanges()</a></code> function, based on a reference GTF file. Here, we exemplify this with a small subset of a GTF file from <a href="https://www.gencodegenes.org/human/release_28.html">Gencode release 28</a>. We extract genomic ranges for spliced transcript and introns, where the latter are defined for each transcript separately (using the same terminology as in the <em><a href="https://bioconductor.org/packages/3.11/BUSpaRse">BUSpaRse</a></em> package). For each intron, a flanking sequence of 50 nt is added on each side to accommodate reads mapping across an exon/intron boundary.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">gtf &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/gencode.v28.annotation.sub.gtf"</span>, </a>
<a class="sourceLine" id="cb2-2" data-line-number="2">                   <span class="dt">package =</span> <span class="st">"eisaR"</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">grl &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getFeatureRanges.html">getFeatureRanges</a></span>(</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">  <span class="dt">gtf =</span> gtf,</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  <span class="dt">featureType =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"spliced"</span>, <span class="st">"intron"</span>), </a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  <span class="dt">intronType =</span> <span class="st">"separate"</span>, </a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="dt">flankLength =</span> 50L, </a>
<a class="sourceLine" id="cb2-8" data-line-number="8">  <span class="dt">joinOverlappingIntrons =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb2-9" data-line-number="9">  <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10">)</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt; Import genomic features from the file as a GRanges object ... OK</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#&gt; Prepare the 'metadata' data frame ... OK</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co">#&gt; Make the TxDb object ...</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt;   features of type stop_codon. This information was ignored.</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt; OK</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt; Extracting spliced transcript features</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="co">#&gt; Extracting introns using the separate approach</span></a></code></pre></div>
<p>The output from <code><a href="../reference/getFeatureRanges.html">getFeatureRanges()</a></code> is a <code>GRangesList</code> object, with all the features of interest (both spliced transcripts and introns):</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">grl</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; GRangesList object of length 895:</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; $ENST00000456328.2</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; GRanges object with 3 ranges and 5 metadata columns:</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt;       seqnames      ranges strand |           exon_id exon_rank</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;character&gt; &lt;integer&gt;</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt;   [1]     chr1 11869-12227      + | ENSE00002234944.1         1</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;   [2]     chr1 12613-12721      + | ENSE00003582793.1         2</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;   [3]     chr1 13221-14409      + | ENSE00002312635.1         3</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt;           transcript_id           gene_id        type</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;             &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;   [1] ENST00000456328.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;   [2] ENST00000456328.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;   [3] ENST00000456328.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt;   -------</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; $ENST00000450305.2</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; GRanges object with 6 ranges and 5 metadata columns:</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt;       seqnames      ranges strand |           exon_id exon_rank</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;character&gt; &lt;integer&gt;</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt;   [1]     chr1 12010-12057      + | ENSE00001948541.1         1</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt;   [2]     chr1 12179-12227      + | ENSE00001671638.2         2</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt;   [3]     chr1 12613-12697      + | ENSE00001758273.2         3</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt;   [4]     chr1 12975-13052      + | ENSE00001799933.2         4</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt;   [5]     chr1 13221-13374      + | ENSE00001746346.2         5</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt;   [6]     chr1 13453-13670      + | ENSE00001863096.1         6</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt;           transcript_id           gene_id        type</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt;             &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">#&gt;   [1] ENST00000450305.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">#&gt;   [2] ENST00000450305.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">#&gt;   [3] ENST00000450305.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="co">#&gt;   [4] ENST00000450305.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="co">#&gt;   [5] ENST00000450305.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="co">#&gt;   [6] ENST00000450305.2 ENSG00000223972.5        exon</span></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"><span class="co">#&gt;   -------</span></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"><span class="co">#&gt; $ENST00000473358.1</span></a>
<a class="sourceLine" id="cb3-40" data-line-number="40"><span class="co">#&gt; GRanges object with 3 ranges and 5 metadata columns:</span></a>
<a class="sourceLine" id="cb3-41" data-line-number="41"><span class="co">#&gt;       seqnames      ranges strand |           exon_id exon_rank</span></a>
<a class="sourceLine" id="cb3-42" data-line-number="42"><span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;character&gt; &lt;integer&gt;</span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43"><span class="co">#&gt;   [1]     chr1 29554-30039      + | ENSE00001947070.1         1</span></a>
<a class="sourceLine" id="cb3-44" data-line-number="44"><span class="co">#&gt;   [2]     chr1 30564-30667      + | ENSE00001922571.1         2</span></a>
<a class="sourceLine" id="cb3-45" data-line-number="45"><span class="co">#&gt;   [3]     chr1 30976-31097      + | ENSE00001827679.1         3</span></a>
<a class="sourceLine" id="cb3-46" data-line-number="46"><span class="co">#&gt;           transcript_id           gene_id        type</span></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"><span class="co">#&gt;             &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span></a>
<a class="sourceLine" id="cb3-48" data-line-number="48"><span class="co">#&gt;   [1] ENST00000473358.1 ENSG00000243485.5        exon</span></a>
<a class="sourceLine" id="cb3-49" data-line-number="49"><span class="co">#&gt;   [2] ENST00000473358.1 ENSG00000243485.5        exon</span></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="co">#&gt;   [3] ENST00000473358.1 ENSG00000243485.5        exon</span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="co">#&gt;   -------</span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></a>
<a class="sourceLine" id="cb3-53" data-line-number="53"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-54" data-line-number="54"><span class="co">#&gt; ...</span></a>
<a class="sourceLine" id="cb3-55" data-line-number="55"><span class="co">#&gt; &lt;892 more elements&gt;</span></a></code></pre></div>
<p>The <code>metadata</code> slot of the <code>GRangesList</code> object contains a list of the feature IDs for each type of feature:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(S4Vectors<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata</a></span>(grl)<span class="op">$</span>featurelist, head)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt; $spliced</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; [1] "ENST00000456328.2" "ENST00000450305.2" "ENST00000473358.1"</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; [4] "ENST00000469289.1" "ENST00000607096.1" "ENST00000606857.1"</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; $intron</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; [1] "ENST00000456328.2-I"  "ENST00000456328.2-I1"</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; [3] "ENST00000450305.2-I"  "ENST00000450305.2-I1"</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; [5] "ENST00000450305.2-I2" "ENST00000450305.2-I3"</span></a></code></pre></div>
<p>As we can see, the intron IDs are identified by a <code>-I</code> suffix. Each feature is further annotated to a gene ID. For the intronic features, the corresponding gene ID also bears the <code>-I</code> suffix appended to the original gene ID. Having separate gene IDs for spliced transcripts and introns allows, for example, joint quantification of spliced and unspliced versions of a gene with alevin. Adding a suffix rather than defining a completely new gene ID allows us to easily match corresponding spliced and unspliced feature IDs. To further simplify the latter, the metadata of the <code>GRangesList</code> object returned by <code><a href="../reference/getFeatureRanges.html">getFeatureRanges()</a></code> contains <code>data.frame</code>s matching the corresponding gene IDs (as well as transcript IDs, if unspliced transcripts are extracted):</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(S4Vectors<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata</a></span>(grl)<span class="op">$</span>corrgene)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt;             spliced              intron</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; 1 ENSG00000223972.5 ENSG00000223972.5-I</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; 2 ENSG00000243485.5 ENSG00000243485.5-I</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; 3 ENSG00000284332.1 ENSG00000284332.1-I</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; 4 ENSG00000268020.3 ENSG00000268020.3-I</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; 5 ENSG00000240361.2 ENSG00000240361.2-I</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; 6 ENSG00000186092.6 ENSG00000186092.6-I</span></a></code></pre></div>
</div>
<div id="extract-feature-sequences" class="section level1">
<h1 class="hasAnchor">
<a href="#extract-feature-sequences" class="anchor"></a>Extract feature sequences</h1>
<p>Once the genomic ranges of the features of interest are extracted, we can obtain the nucleotide sequences using the <code>extractTranscriptSeqs()</code> function from the <em><a href="https://bioconductor.org/packages/3.11/GenomicFeatures">GenomicFeatures</a></em> package. In addition to the ranges, this requires the genome sequence. This can be obtained, for example, from the corresponding BSgenome package, or from an external FASTA file.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>({</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BSgenome)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BSgenome.Hsapiens.UCSC.hg38)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">})</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">seqs &lt;-<span class="st"> </span>GenomicFeatures<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/extractTranscriptSeqs.html">extractTranscriptSeqs</a></span>(</a>
<a class="sourceLine" id="cb6-6" data-line-number="6">  <span class="dt">x =</span> BSgenome.Hsapiens.UCSC.hg38, </a>
<a class="sourceLine" id="cb6-7" data-line-number="7">  <span class="dt">transcripts =</span> grl</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">)</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">seqs</a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; DNAStringSet object of length 895:</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;        width seq                                  names               </span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;   [1]   1657 GTTAACTTGCCGTCAGC...ACACTGTTGGTTTCTG ENST00000456328.2</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt;   [2]    632 GTGTCTGACTTCCAGCA...ACAGGGGAATCCCGAA ENST00000450305.2</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;   [3]    712 GTGCACACGGCTCCCAT...TGAGGGATAAATGTAT ENST00000473358.1</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt;   [4]    535 TCATCAGTCCAAAGTCC...GTATGATTTTAAAGTC ENST00000469289.1</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt;   [5]    138 GGATGCCCAGCTAGTTT...AGAATTAAGCATTTTA ENST00000607096.1</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt;   ...    ... ...</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; [891]    178 GCGCCAGCCGGACCCCA...CGCGTATTAACGAGAG ENST00000304952.1...</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt; [892]    193 GGAGATGACCGTGAGAC...GTACCGCGCCGGCTTC ENST00000481869.1-I</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; [893]    193 GGAGATGACCGTGAGAC...GTACCGCGCCGGCTTC ENST00000484667.2-I</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt; [894]    352 GCGCCAGCCGGACCCCA...TCCTGGAGATGACCGT ENST00000484667.2-I1</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; [895]    826 ACCTCCGCCTGCCAGGC...AGAGATTGTGGTGAGC ENST00000458555.1-I</span></a></code></pre></div>
<p>The resulting <code>DNAStringSet</code> can be written to a FASTA file and used to generate an index for alignment-free methods such as <em>Salmon</em> and <em>kallisto</em>.</p>
</div>
<div id="generate-an-expanded-gtf-file" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-an-expanded-gtf-file" class="anchor"></a>Generate an expanded GTF file</h1>
<p>In addition to extracting feature sequences, we can also export a GTF file with the full set of features. This is useful, for example, in order to generate a linked transcriptome for later import of estimated abundances with <em><a href="https://bioconductor.org/packages/3.11/tximeta">tximeta</a></em>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="../reference/exportToGtf.html">exportToGtf</a></span>(</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">  grl, </a>
<a class="sourceLine" id="cb7-3" data-line-number="3">  <span class="dt">filepath =</span> <span class="kw"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"exported.gtf"</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">)</a></code></pre></div>
<p>In the exported GTF file, each entry of <code>grl</code> will correspond to a “transcript” feature, and each individual range corresponds to an “exon” feature. In addition, each gene is represented as a “gene” feature.</p>
</div>
<div id="generate-a-transcript-to-gene-mapping" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-a-transcript-to-gene-mapping" class="anchor"></a>Generate a transcript-to-gene mapping</h1>
<p>Finally, we can export a <code>data.frame</code> (or a tab-separated test file) providing a conversion table between “transcript” and “gene” identifiers. This is needed to aggregate the transcript-level abundance estimates from alignment-free methods such as <em>Salmon</em> and <em>kallisto</em> to the gene level.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getTx2Gene.html">getTx2Gene</a></span>(grl)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(df)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt;                       transcript_id           gene_id</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt; ENST00000456328.2 ENST00000456328.2 ENSG00000223972.5</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt; ENST00000450305.2 ENST00000450305.2 ENSG00000223972.5</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt; ENST00000473358.1 ENST00000473358.1 ENSG00000243485.5</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt; ENST00000469289.1 ENST00000469289.1 ENSG00000243485.5</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt; ENST00000607096.1 ENST00000607096.1 ENSG00000284332.1</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt; ENST00000606857.1 ENST00000606857.1 ENSG00000268020.3</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">tail</a></span>(df)</a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt;                               transcript_id              gene_id</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt; ENST00000304952.10-I1 ENST00000304952.10-I1 ENSG00000188290.10-I</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="co">#&gt; ENST00000304952.10-I2 ENST00000304952.10-I2 ENSG00000188290.10-I</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co">#&gt; ENST00000481869.1-I     ENST00000481869.1-I ENSG00000188290.10-I</span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt; ENST00000484667.2-I     ENST00000484667.2-I ENSG00000188290.10-I</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">#&gt; ENST00000484667.2-I1   ENST00000484667.2-I1 ENSG00000188290.10-I</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="co">#&gt; ENST00000458555.1-I     ENST00000458555.1-I  ENSG00000224969.1-I</span></a></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; R Under development (unstable) (2020-03-13 r77948)</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; Running under: Ubuntu 16.04.6 LTS</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils    </span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="co">#&gt; [7] datasets  methods   base     </span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="co">#&gt;  [1] BSgenome.Hsapiens.UCSC.hg38_1.4.3</span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="co">#&gt;  [2] BSgenome_1.55.4                  </span></a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="co">#&gt;  [3] rtracklayer_1.47.0               </span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="co">#&gt;  [4] Biostrings_2.55.6                </span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27"><span class="co">#&gt;  [5] XVector_0.27.1                   </span></a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="co">#&gt;  [6] GenomicRanges_1.39.2             </span></a>
<a class="sourceLine" id="cb9-29" data-line-number="29"><span class="co">#&gt;  [7] GenomeInfoDb_1.23.13             </span></a>
<a class="sourceLine" id="cb9-30" data-line-number="30"><span class="co">#&gt;  [8] IRanges_2.21.5                   </span></a>
<a class="sourceLine" id="cb9-31" data-line-number="31"><span class="co">#&gt;  [9] S4Vectors_0.25.13                </span></a>
<a class="sourceLine" id="cb9-32" data-line-number="32"><span class="co">#&gt; [10] BiocGenerics_0.33.2              </span></a>
<a class="sourceLine" id="cb9-33" data-line-number="33"><span class="co">#&gt; [11] eisaR_0.99.7                     </span></a>
<a class="sourceLine" id="cb9-34" data-line-number="34"><span class="co">#&gt; [12] BiocStyle_2.15.6                 </span></a>
<a class="sourceLine" id="cb9-35" data-line-number="35"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-36" data-line-number="36"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb9-37" data-line-number="37"><span class="co">#&gt;  [1] Biobase_2.47.3              httr_1.4.1                 </span></a>
<a class="sourceLine" id="cb9-38" data-line-number="38"><span class="co">#&gt;  [3] edgeR_3.29.1                bit64_0.9-7                </span></a>
<a class="sourceLine" id="cb9-39" data-line-number="39"><span class="co">#&gt;  [5] assertthat_0.2.1            askpass_1.1                </span></a>
<a class="sourceLine" id="cb9-40" data-line-number="40"><span class="co">#&gt;  [7] BiocManager_1.30.10         BiocFileCache_1.11.4       </span></a>
<a class="sourceLine" id="cb9-41" data-line-number="41"><span class="co">#&gt;  [9] blob_1.2.1                  GenomeInfoDbData_1.2.2     </span></a>
<a class="sourceLine" id="cb9-42" data-line-number="42"><span class="co">#&gt; [11] Rsamtools_2.3.7             yaml_2.2.1                 </span></a>
<a class="sourceLine" id="cb9-43" data-line-number="43"><span class="co">#&gt; [13] progress_1.2.2              pillar_1.4.3               </span></a>
<a class="sourceLine" id="cb9-44" data-line-number="44"><span class="co">#&gt; [15] RSQLite_2.2.0               backports_1.1.5            </span></a>
<a class="sourceLine" id="cb9-45" data-line-number="45"><span class="co">#&gt; [17] lattice_0.20-40             glue_1.3.2                 </span></a>
<a class="sourceLine" id="cb9-46" data-line-number="46"><span class="co">#&gt; [19] limma_3.43.5                digest_0.6.25              </span></a>
<a class="sourceLine" id="cb9-47" data-line-number="47"><span class="co">#&gt; [21] htmltools_0.4.0             Matrix_1.2-18              </span></a>
<a class="sourceLine" id="cb9-48" data-line-number="48"><span class="co">#&gt; [23] XML_3.99-0.3                pkgconfig_2.0.3            </span></a>
<a class="sourceLine" id="cb9-49" data-line-number="49"><span class="co">#&gt; [25] biomaRt_2.43.3              bookdown_0.18              </span></a>
<a class="sourceLine" id="cb9-50" data-line-number="50"><span class="co">#&gt; [27] zlibbioc_1.33.1             purrr_0.3.3                </span></a>
<a class="sourceLine" id="cb9-51" data-line-number="51"><span class="co">#&gt; [29] BiocParallel_1.21.2         tibble_2.1.3               </span></a>
<a class="sourceLine" id="cb9-52" data-line-number="52"><span class="co">#&gt; [31] openssl_1.4.1               SummarizedExperiment_1.17.3</span></a>
<a class="sourceLine" id="cb9-53" data-line-number="53"><span class="co">#&gt; [33] GenomicFeatures_1.39.7      magrittr_1.5               </span></a>
<a class="sourceLine" id="cb9-54" data-line-number="54"><span class="co">#&gt; [35] crayon_1.3.4                memoise_1.1.0              </span></a>
<a class="sourceLine" id="cb9-55" data-line-number="55"><span class="co">#&gt; [37] evaluate_0.14               fs_1.3.1                   </span></a>
<a class="sourceLine" id="cb9-56" data-line-number="56"><span class="co">#&gt; [39] MASS_7.3-51.5               tools_4.0.0                </span></a>
<a class="sourceLine" id="cb9-57" data-line-number="57"><span class="co">#&gt; [41] prettyunits_1.1.1           hms_0.5.3                  </span></a>
<a class="sourceLine" id="cb9-58" data-line-number="58"><span class="co">#&gt; [43] matrixStats_0.56.0          stringr_1.4.0              </span></a>
<a class="sourceLine" id="cb9-59" data-line-number="59"><span class="co">#&gt; [45] locfit_1.5-9.1              DelayedArray_0.13.7        </span></a>
<a class="sourceLine" id="cb9-60" data-line-number="60"><span class="co">#&gt; [47] AnnotationDbi_1.49.1        compiler_4.0.0             </span></a>
<a class="sourceLine" id="cb9-61" data-line-number="61"><span class="co">#&gt; [49] pkgdown_1.4.1               rlang_0.4.5                </span></a>
<a class="sourceLine" id="cb9-62" data-line-number="62"><span class="co">#&gt; [51] grid_4.0.0                  RCurl_1.98-1.1             </span></a>
<a class="sourceLine" id="cb9-63" data-line-number="63"><span class="co">#&gt; [53] rappdirs_0.3.1              bitops_1.0-6               </span></a>
<a class="sourceLine" id="cb9-64" data-line-number="64"><span class="co">#&gt; [55] rmarkdown_2.1               DBI_1.1.0                  </span></a>
<a class="sourceLine" id="cb9-65" data-line-number="65"><span class="co">#&gt; [57] curl_4.3                    R6_2.4.1                   </span></a>
<a class="sourceLine" id="cb9-66" data-line-number="66"><span class="co">#&gt; [59] GenomicAlignments_1.23.1    knitr_1.28                 </span></a>
<a class="sourceLine" id="cb9-67" data-line-number="67"><span class="co">#&gt; [61] dplyr_0.8.5                 bit_1.1-15.2               </span></a>
<a class="sourceLine" id="cb9-68" data-line-number="68"><span class="co">#&gt; [63] rprojroot_1.3-2             desc_1.2.0                 </span></a>
<a class="sourceLine" id="cb9-69" data-line-number="69"><span class="co">#&gt; [65] stringi_1.4.6               Rcpp_1.0.4                 </span></a>
<a class="sourceLine" id="cb9-70" data-line-number="70"><span class="co">#&gt; [67] vctrs_0.2.4                 dbplyr_1.4.2               </span></a>
<a class="sourceLine" id="cb9-71" data-line-number="71"><span class="co">#&gt; [69] tidyselect_1.0.0            xfun_0.12</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#generate-feature-ranges">Generate feature ranges</a></li>
      <li><a href="#extract-feature-sequences">Extract feature sequences</a></li>
      <li><a href="#generate-an-expanded-gtf-file">Generate an expanded GTF file</a></li>
      <li><a href="#generate-a-transcript-to-gene-mapping">Generate a transcript-to-gene mapping</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/mbstadler">Michael Stadler</a>, Dimos Gaidatzis, <a href="https://github.com/LukasBurger">Lukas Burger</a>, <a href="https://csoneson.github.io">Charlotte Soneson</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
