<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Generating reference files for spliced and unspliced abundance estimation with alignment-free methods • eisaR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Generating reference files for spliced and unspliced abundance estimation with alignment-free methods">
<meta property="og:description" content="eisaR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">eisaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/eisaR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/eisaR.html">eisaR</a>
    </li>
    <li>
      <a href="../articles/rna-velocity.html">Feature extraction for RNA velocity</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fmicompbio/eisaR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Generating reference files for spliced and unspliced abundance estimation with alignment-free methods</h1>
                        <h4 class="author">Charlotte Soneson</h4>
            
            <h4 class="date">2020-03-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/fmicompbio/eisaR/blob/master/vignettes/rna-velocity.Rmd"><code>vignettes/rna-velocity.Rmd</code></a></small>
      <div class="hidden name"><code>rna-velocity.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In this vignette, we show how to prepare reference files for estimating abundances of spliced and unspliced abundances with alignment-free methods (e.g., <a href="https://salmon.readthedocs.io/en/latest/salmon.html">Salmon</a>, <a href="https://salmon.readthedocs.io/en/latest/alevin.html">alevin</a> or <a href="https://pachterlab.github.io/kallisto/">kallisto</a>). Such abundances are used as input, e.g., for estimation of RNA velocity in single-cell data.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">eisaR</span>)</pre></body></html></div>
</div>
<div id="generate-feature-ranges" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-feature-ranges" class="anchor"></a>Generate feature ranges</h1>
<p>The first step is to generate a <code>GRangesList</code> object containing the genomic ranges for the features of interest (spliced transcripts, and either unspliced transcripts or intron sequences). This is done using the <code><a href="../reference/getFeatureRanges.html">getFeatureRanges()</a></code> function, based on a reference GTF file. Here, we exemplify this with a small subset of a GTF file from <a href="https://www.gencodegenes.org/human/release_28.html">Gencode release 28</a>. We extract genomic ranges for spliced transcript and introns, where the latter are defined for each transcript separately (using the same terminology as in the <em><a href="https://bioconductor.org/packages/3.11/BUSpaRse">BUSpaRse</a></em> package). For each intron, a flanking sequence of 50 nt is added on each side to accommodate reads mapping across an exon/intron boundary.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">gtf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/gencode.v28.annotation.sub.gtf"</span>,
                   <span class="kw">package</span> <span class="kw">=</span> <span class="st">"eisaR"</span>)
<span class="no">grl</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getFeatureRanges.html">getFeatureRanges</a></span>(
  <span class="kw">gtf</span> <span class="kw">=</span> <span class="no">gtf</span>,
  <span class="kw">featureType</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"spliced"</span>, <span class="st">"intron"</span>),
  <span class="kw">intronType</span> <span class="kw">=</span> <span class="st">"separate"</span>,
  <span class="kw">flankLength</span> <span class="kw">=</span> <span class="fl">50L</span>,
  <span class="kw">joinOverlappingIntrons</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)
<span class="co">#&gt; Import genomic features from the file as a GRanges object ... OK</span>
<span class="co">#&gt; Prepare the 'metadata' data frame ... OK</span>
<span class="co">#&gt; Make the TxDb object ...</span>
<span class="co">#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for</span>
<span class="co">#&gt;   features of type stop_codon. This information was ignored.</span>
<span class="co">#&gt; OK</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; Extracting spliced transcript features</span>
<span class="co">#&gt; Extracting introns using the separate approach</span></pre></body></html></div>
<p>The output from <code><a href="../reference/getFeatureRanges.html">getFeatureRanges()</a></code> is a <code>GRangesList</code> object, with all the features of interest (both spliced transcripts and introns):</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">grl</span>
<span class="co">#&gt; GRangesList object of length 895:</span>
<span class="co">#&gt; $ENST00000456328.2</span>
<span class="co">#&gt; GRanges object with 3 ranges and 5 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand |           exon_id exon_rank</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   [1]     chr1 11869-12227      + | ENSE00002234944.1         1</span>
<span class="co">#&gt;   [2]     chr1 12613-12721      + | ENSE00003582793.1         2</span>
<span class="co">#&gt;   [3]     chr1 13221-14409      + | ENSE00002312635.1         3</span>
<span class="co">#&gt;           transcript_id           gene_id        type</span>
<span class="co">#&gt;             &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1] ENST00000456328.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   [2] ENST00000456328.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   [3] ENST00000456328.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ENST00000450305.2</span>
<span class="co">#&gt; GRanges object with 6 ranges and 5 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand |           exon_id exon_rank</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   [1]     chr1 12010-12057      + | ENSE00001948541.1         1</span>
<span class="co">#&gt;   [2]     chr1 12179-12227      + | ENSE00001671638.2         2</span>
<span class="co">#&gt;   [3]     chr1 12613-12697      + | ENSE00001758273.2         3</span>
<span class="co">#&gt;   [4]     chr1 12975-13052      + | ENSE00001799933.2         4</span>
<span class="co">#&gt;   [5]     chr1 13221-13374      + | ENSE00001746346.2         5</span>
<span class="co">#&gt;   [6]     chr1 13453-13670      + | ENSE00001863096.1         6</span>
<span class="co">#&gt;           transcript_id           gene_id        type</span>
<span class="co">#&gt;             &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1] ENST00000450305.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   [2] ENST00000450305.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   [3] ENST00000450305.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   [4] ENST00000450305.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   [5] ENST00000450305.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   [6] ENST00000450305.2 ENSG00000223972.5        exon</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ENST00000473358.1</span>
<span class="co">#&gt; GRanges object with 3 ranges and 5 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand |           exon_id exon_rank</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   [1]     chr1 29554-30039      + | ENSE00001947070.1         1</span>
<span class="co">#&gt;   [2]     chr1 30564-30667      + | ENSE00001922571.1         2</span>
<span class="co">#&gt;   [3]     chr1 30976-31097      + | ENSE00001827679.1         3</span>
<span class="co">#&gt;           transcript_id           gene_id        type</span>
<span class="co">#&gt;             &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1] ENST00000473358.1 ENSG00000243485.5        exon</span>
<span class="co">#&gt;   [2] ENST00000473358.1 ENSG00000243485.5        exon</span>
<span class="co">#&gt;   [3] ENST00000473358.1 ENSG00000243485.5        exon</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; &lt;892 more elements&gt;</span></pre></body></html></div>
<p>The <code>metadata</code> slot of the <code>GRangesList</code> object contains a list of the feature IDs for each type of feature:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw pkg">S4Vectors</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata</a></span>(<span class="no">grl</span>)$<span class="no">featurelist</span>, <span class="no">head</span>)
<span class="co">#&gt; $spliced</span>
<span class="co">#&gt; [1] "ENST00000456328.2" "ENST00000450305.2" "ENST00000473358.1"</span>
<span class="co">#&gt; [4] "ENST00000469289.1" "ENST00000607096.1" "ENST00000606857.1"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $intron</span>
<span class="co">#&gt; [1] "ENST00000456328.2-I"  "ENST00000456328.2-I1"</span>
<span class="co">#&gt; [3] "ENST00000450305.2-I"  "ENST00000450305.2-I1"</span>
<span class="co">#&gt; [5] "ENST00000450305.2-I2" "ENST00000450305.2-I3"</span></pre></body></html></div>
<p>As we can see, the intron IDs are identified by a <code>-I</code> suffix. Each feature is further annotated to a gene ID. For the intronic features, the corresponding gene ID also bears the <code>-I</code> suffix appended to the original gene ID. Having separate gene IDs for spliced transcripts and introns allows, for example, joint quantification of spliced and unspliced versions of a gene with alevin. Adding a suffix rather than defining a completely new gene ID allows us to easily match corresponding spliced and unspliced feature IDs. To further simplify the latter, the metadata of the <code>GRangesList</code> object returned by <code><a href="../reference/getFeatureRanges.html">getFeatureRanges()</a></code> contains <code>data.frame</code>s matching the corresponding gene IDs (as well as transcript IDs, if unspliced transcripts are extracted):</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw pkg">S4Vectors</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html">metadata</a></span>(<span class="no">grl</span>)$<span class="no">corrgene</span>)
<span class="co">#&gt;             spliced              intron</span>
<span class="co">#&gt; 1 ENSG00000223972.5 ENSG00000223972.5-I</span>
<span class="co">#&gt; 2 ENSG00000243485.5 ENSG00000243485.5-I</span>
<span class="co">#&gt; 3 ENSG00000284332.1 ENSG00000284332.1-I</span>
<span class="co">#&gt; 4 ENSG00000268020.3 ENSG00000268020.3-I</span>
<span class="co">#&gt; 5 ENSG00000240361.2 ENSG00000240361.2-I</span>
<span class="co">#&gt; 6 ENSG00000186092.6 ENSG00000186092.6-I</span></pre></body></html></div>
</div>
<div id="extract-feature-sequences" class="section level1">
<h1 class="hasAnchor">
<a href="#extract-feature-sequences" class="anchor"></a>Extract feature sequences</h1>
<p>Once the genomic ranges of the features of interest are extracted, we can obtain the nucleotide sequences using the <code>extractTranscriptSeqs()</code> function from the <em><a href="https://bioconductor.org/packages/3.11/GenomicFeatures">GenomicFeatures</a></em> package. In addition to the ranges, this requires the genome sequence. This can be obtained, for example, from the corresponding BSgenome package, or from an external FASTA file.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>({
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BSgenome</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BSgenome.Hsapiens.UCSC.hg38</span>)
})
<span class="no">seqs</span> <span class="kw">&lt;-</span> <span class="kw pkg">GenomicFeatures</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/extractTranscriptSeqs.html">extractTranscriptSeqs</a></span>(
  <span class="kw">x</span> <span class="kw">=</span> <span class="no">BSgenome.Hsapiens.UCSC.hg38</span>,
  <span class="kw">transcripts</span> <span class="kw">=</span> <span class="no">grl</span>
)
<span class="no">seqs</span>
<span class="co">#&gt; DNAStringSet object of length 895:</span>
<span class="co">#&gt;        width seq                                  names               </span>
<span class="co">#&gt;   [1]   1657 GTTAACTTGCCGTCAGC...ACACTGTTGGTTTCTG ENST00000456328.2</span>
<span class="co">#&gt;   [2]    632 GTGTCTGACTTCCAGCA...ACAGGGGAATCCCGAA ENST00000450305.2</span>
<span class="co">#&gt;   [3]    712 GTGCACACGGCTCCCAT...TGAGGGATAAATGTAT ENST00000473358.1</span>
<span class="co">#&gt;   [4]    535 TCATCAGTCCAAAGTCC...GTATGATTTTAAAGTC ENST00000469289.1</span>
<span class="co">#&gt;   [5]    138 GGATGCCCAGCTAGTTT...AGAATTAAGCATTTTA ENST00000607096.1</span>
<span class="co">#&gt;   ...    ... ...</span>
<span class="co">#&gt; [891]    178 GCGCCAGCCGGACCCCA...CGCGTATTAACGAGAG ENST00000304952.1...</span>
<span class="co">#&gt; [892]    193 GGAGATGACCGTGAGAC...GTACCGCGCCGGCTTC ENST00000481869.1-I</span>
<span class="co">#&gt; [893]    193 GGAGATGACCGTGAGAC...GTACCGCGCCGGCTTC ENST00000484667.2-I</span>
<span class="co">#&gt; [894]    352 GCGCCAGCCGGACCCCA...TCCTGGAGATGACCGT ENST00000484667.2-I1</span>
<span class="co">#&gt; [895]    826 ACCTCCGCCTGCCAGGC...AGAGATTGTGGTGAGC ENST00000458555.1-I</span></pre></body></html></div>
<p>The resulting <code>DNAStringSet</code> can be written to a FASTA file and used to generate an index for alignment-free methods such as <em>Salmon</em> and <em>kallisto</em>.</p>
</div>
<div id="generate-an-expanded-gtf-file" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-an-expanded-gtf-file" class="anchor"></a>Generate an expanded GTF file</h1>
<p>In addition to extracting feature sequences, we can also export a GTF file with the full set of features. This is useful, for example, in order to generate a linked transcriptome for later import of estimated abundances with <em><a href="https://bioconductor.org/packages/3.11/tximeta">tximeta</a></em>.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="../reference/exportToGtf.html">exportToGtf</a></span>(
  <span class="no">grl</span>,
  <span class="kw">filepath</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"exported.gtf"</span>)
)</pre></body></html></div>
<p>In the exported GTF file, each entry of <code>grl</code> will correspond to a “transcript” feature, and each individual range corresponds to an “exon” feature. In addition, each gene is represented as a “gene” feature.</p>
</div>
<div id="generate-a-transcript-to-gene-mapping" class="section level1">
<h1 class="hasAnchor">
<a href="#generate-a-transcript-to-gene-mapping" class="anchor"></a>Generate a transcript-to-gene mapping</h1>
<p>Finally, we can export a <code>data.frame</code> (or a tab-separated test file) providing a conversion table between “transcript” and “gene” identifiers. This is needed to aggregate the transcript-level abundance estimates from alignment-free methods such as <em>Salmon</em> and <em>kallisto</em> to the gene level.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getTx2Gene.html">getTx2Gene</a></span>(<span class="no">grl</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">df</span>)
<span class="co">#&gt;                       transcript_id           gene_id</span>
<span class="co">#&gt; ENST00000456328.2 ENST00000456328.2 ENSG00000223972.5</span>
<span class="co">#&gt; ENST00000450305.2 ENST00000450305.2 ENSG00000223972.5</span>
<span class="co">#&gt; ENST00000473358.1 ENST00000473358.1 ENSG00000243485.5</span>
<span class="co">#&gt; ENST00000469289.1 ENST00000469289.1 ENSG00000243485.5</span>
<span class="co">#&gt; ENST00000607096.1 ENST00000607096.1 ENSG00000284332.1</span>
<span class="co">#&gt; ENST00000606857.1 ENST00000606857.1 ENSG00000268020.3</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span>(<span class="no">df</span>)
<span class="co">#&gt;                               transcript_id              gene_id</span>
<span class="co">#&gt; ENST00000304952.10-I1 ENST00000304952.10-I1 ENSG00000188290.10-I</span>
<span class="co">#&gt; ENST00000304952.10-I2 ENST00000304952.10-I2 ENSG00000188290.10-I</span>
<span class="co">#&gt; ENST00000481869.1-I     ENST00000481869.1-I ENSG00000188290.10-I</span>
<span class="co">#&gt; ENST00000484667.2-I     ENST00000484667.2-I ENSG00000188290.10-I</span>
<span class="co">#&gt; ENST00000484667.2-I1   ENST00000484667.2-I1 ENSG00000188290.10-I</span>
<span class="co">#&gt; ENST00000458555.1-I     ENST00000458555.1-I  ENSG00000224969.1-I</span></pre></body></html></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R Under development (unstable) (2020-03-25 r78063)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils    </span>
<span class="co">#&gt; [7] datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] BSgenome.Hsapiens.UCSC.hg38_1.4.3</span>
<span class="co">#&gt;  [2] BSgenome_1.55.4                  </span>
<span class="co">#&gt;  [3] rtracklayer_1.47.0               </span>
<span class="co">#&gt;  [4] Biostrings_2.55.7                </span>
<span class="co">#&gt;  [5] XVector_0.27.2                   </span>
<span class="co">#&gt;  [6] GenomicRanges_1.39.3             </span>
<span class="co">#&gt;  [7] GenomeInfoDb_1.23.16             </span>
<span class="co">#&gt;  [8] IRanges_2.21.8                   </span>
<span class="co">#&gt;  [9] S4Vectors_0.25.14                </span>
<span class="co">#&gt; [10] BiocGenerics_0.33.3              </span>
<span class="co">#&gt; [11] eisaR_0.99.10                    </span>
<span class="co">#&gt; [12] BiocStyle_2.15.6                 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Biobase_2.47.3              httr_1.4.1                 </span>
<span class="co">#&gt;  [3] edgeR_3.29.1                bit64_0.9-7                </span>
<span class="co">#&gt;  [5] assertthat_0.2.1            askpass_1.1                </span>
<span class="co">#&gt;  [7] BiocManager_1.30.10         BiocFileCache_1.11.4       </span>
<span class="co">#&gt;  [9] blob_1.2.1                  GenomeInfoDbData_1.2.2     </span>
<span class="co">#&gt; [11] Rsamtools_2.3.7             yaml_2.2.1                 </span>
<span class="co">#&gt; [13] progress_1.2.2              pillar_1.4.3               </span>
<span class="co">#&gt; [15] RSQLite_2.2.0               backports_1.1.5            </span>
<span class="co">#&gt; [17] lattice_0.20-40             glue_1.3.2                 </span>
<span class="co">#&gt; [19] limma_3.43.5                digest_0.6.25              </span>
<span class="co">#&gt; [21] htmltools_0.4.0             Matrix_1.2-18              </span>
<span class="co">#&gt; [23] XML_3.99-0.3                pkgconfig_2.0.3            </span>
<span class="co">#&gt; [25] biomaRt_2.43.4              bookdown_0.18              </span>
<span class="co">#&gt; [27] zlibbioc_1.33.1             purrr_0.3.3                </span>
<span class="co">#&gt; [29] BiocParallel_1.21.2         tibble_3.0.0               </span>
<span class="co">#&gt; [31] openssl_1.4.1               ellipsis_0.3.0             </span>
<span class="co">#&gt; [33] SummarizedExperiment_1.17.5 GenomicFeatures_1.39.7     </span>
<span class="co">#&gt; [35] cli_2.0.2                   magrittr_1.5               </span>
<span class="co">#&gt; [37] crayon_1.3.4                memoise_1.1.0              </span>
<span class="co">#&gt; [39] evaluate_0.14               fs_1.3.2                   </span>
<span class="co">#&gt; [41] fansi_0.4.1                 MASS_7.3-51.5              </span>
<span class="co">#&gt; [43] tools_4.0.0                 prettyunits_1.1.1          </span>
<span class="co">#&gt; [45] hms_0.5.3                   lifecycle_0.2.0            </span>
<span class="co">#&gt; [47] matrixStats_0.56.0          stringr_1.4.0              </span>
<span class="co">#&gt; [49] locfit_1.5-9.4              DelayedArray_0.13.8        </span>
<span class="co">#&gt; [51] AnnotationDbi_1.49.1        compiler_4.0.0             </span>
<span class="co">#&gt; [53] pkgdown_1.5.0.9000          rlang_0.4.5                </span>
<span class="co">#&gt; [55] grid_4.0.0                  RCurl_1.98-1.1             </span>
<span class="co">#&gt; [57] rappdirs_0.3.1              bitops_1.0-6               </span>
<span class="co">#&gt; [59] rmarkdown_2.1               DBI_1.1.0                  </span>
<span class="co">#&gt; [61] curl_4.3                    R6_2.4.1                   </span>
<span class="co">#&gt; [63] GenomicAlignments_1.23.2    knitr_1.28                 </span>
<span class="co">#&gt; [65] dplyr_0.8.5                 bit_1.1-15.2               </span>
<span class="co">#&gt; [67] rprojroot_1.3-2             desc_1.2.0                 </span>
<span class="co">#&gt; [69] stringi_1.4.6               Rcpp_1.0.4.4               </span>
<span class="co">#&gt; [71] vctrs_0.2.4                 dbplyr_1.4.2               </span>
<span class="co">#&gt; [73] tidyselect_1.0.0            xfun_0.12</span></pre></body></html></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/mbstadler">Michael Stadler</a>, Dimos Gaidatzis, <a href="https://github.com/LukasBurger">Lukas Burger</a>, <a href="https://csoneson.github.io">Charlotte Soneson</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.0.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
